### DISCUSSION TITLE
How to block update of foreign key constraints

### DISCUSSION THREAD

**Discussion by TannerS** [sequelize/sequelize#14995] [Help & Questions] [answered]
## Issue Creation Checklist

[ X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description
I have a model that corresponds to a entry in our database or when I find a result that matches a model, I am able to use the Model.update method to update that model (aka the DB entry). What I am wondering about is is there anyway to block a foreign key constraint during the update method.

For example let's say I have a model update payload like

```
{
     testField: "test data",
     fooId: 25, // some one or many relationship FK
}
```

What I found it I'm able to pass a payload to change the fooId and I'm wondering if there is some safety method to prevent relationships from being changed?

Thank you

**Comment by ephys** [comment, marked as answer]
You can add hooks, such as `beforeBulkUpdate` & `beforeUpdate` to run arbitrary code, such as checking that fooId hasn't changed https://sequelize.org/docs/v6/other-topics/hooks/
